src/V3AstNodes.cpp

```diff
@@ -2857,6 +2857,152 @@ AstAlways* AstAssignW::convertToAlways() {
     replaceWith(newp);  // User expected to then deleteTree();
     return newp;
 }
+static AstNode* createForeachLoop(AstNodeForeach* nodep, AstNode* bodysp, AstVar* varp,
+                                  AstNodeExpr* leftp, AstNodeExpr* rightp, VNType nodeType) {
```

Comment:
I generally don't like transform code to be in V3AstNodes, this is a low-level header, and if we started down this V3AstNodes would have half of the code in Verilator ;).  Is there some other place this could live - e.g. back as V3Begin::createForeachLoop?

Reply:
Moved back to V3Begin.